### DISCUSSION TITLE
EPS to JPG Poor quality

### DISCUSSION THREAD

**Discussion by DaSwan** [ImageMagick/ImageMagick#3477] [Help]
Hi all !

I need your help to convert EPS to JPG. I would like to convert an EPS file to a 500x500 jpg file. The EPS is natively at 500x200 so the JPG should be of good quality when output. Instead, I grab a blurry JPG and think it converts to a small size JPG first and then enlarges the image.
Here is the command used and the images:
convert -background white -quality 100 -density 500 "'. $ myfile.'" [0] -layers merge -intent relative -black-point-compensation -alpha transparent -clip -alpha opaque -resize '.500'x'. 500. "-flatten -strip" '. $ Directory.' "', $ Myfile.jpg,

[578.zip](https://github.com/ImageMagick/ImageMagick/files/6232527/578.zip)

Thank you in advance !

Regards,

Matt

**Comment by snibgo** [comment]
What version of IM, on what platform? What version of Ghostscript?

```
-resize '.500'x'. 500.
```
That is junk. Perhaps you intended:
```
-resize 500x500
```

**Reply by DaSwan** [reply]
Hi,

i'm running this on Debian Buster with these versions : 

imagemagick                       6.9.10.23+dfsg-2.1+deb10u1
ghostscript                           9.27~dfsg-2+deb10u4


Sorry for the command, the original one is 

convert -background white -quality 100 -density 500  "'.$ori.'"[0] -layers merge -intent relative -black-point-compensation -alpha transparent -clip -alpha opaque -resize '.$lm.'x'.$hm.' -flatten -strip "'.$des.'"', $out, $ret 

with 
$ori = original eps file
$lm = 500
$hm = 500
$des = output folder
$out = output jpg file

Regards,

**Comment by snibgo** [comment]
Please show the actual command that was executed, with all variables substituted.

"-intent relative" and "-black-point-compensation" are relevant only when you are converting with profiles.

Making the background transparent, then flattening, is pointless. And JPG can't record transparency.

"-clip" fails for yor image because it has no clip path.

With IM v6.9.9-50, this command (Windows BAT syntax)...
```
convert ^
   -background white -quality 100 -density 500 ^
  578.eps[0] ^
  -layers merge -intent relative -black-point-compensation ^
  -alpha transparent -clip -alpha opaque -resize 500x500 -flatten -strip ^
  out.jpg
```
... gives this image:
![out](https://user-images.githubusercontent.com/33812368/113122203-4c758d80-920b-11eb-9f9f-7df234c18272.jpg)

This seems good to me.

**Reply by DaSwan** [reply]
Snibgo, thx for your answer.
i'll update my IM version and try with our command to see the result.

Keep in touch
